src/netaddress.cpp

```diff
@@ -537,23 +542,29 @@ std::vector<unsigned char> CNetAddr::GetGroup(const std::vector<bool> &asmap) co
         nBits = 32;
     }
 
-    // push our ip onto vchRet byte by byte...
-    while (nBits >= 8)
-    {
-        vchRet.push_back(GetByte(15 - nStartByte));
-        nStartByte++;
-        nBits -= 8;
-    }
+    // Push our address onto vchRet.
+    const size_t i = nBits >> 3;
+    vchRet.insert(vchRet.end(), m_addr.begin(), m_addr.begin() + i);
+    nBits &= 0x7;
     // ...for the last byte, push nBits and for the rest of the byte push 1's
-    if (nBits > 0)
-        vchRet.push_back(GetByte(15 - nStartByte) | ((1 << (8 - nBits)) - 1));
+    if (nBits > 0) {
+        assert(i < m_addr.size());
```

Comment:
re: https://github.com/bitcoin/bitcoin/pull/19628#discussion_r474617351

> It is here because here we access `m_addr[i]`. I would normally use `m_addr.at(i)` if there is a chance it may get out of bounds and not write an explicit separate assert, but `prevector` does not have an `at()` method :/

Right, but if you assume the assert is false and `num_bytes >= m_addr.size()`, then the expression a few lines earlier `m_addr.begin() + num_bytes` will read from uninitialized memory. So I'm saying it might make sense to move the assert earlier there. Not very important, though.